vertx-mysql-client/src/main/java/io/vertx/mysqlclient/impl/codec/ExtendedBatchQueryCommandCodec.java

```diff
@@ -22,79 +22,106 @@
 
 import java.util.List;
 
-import static io.vertx.mysqlclient.impl.protocol.Packets.EnumCursorType.CURSOR_TYPE_NO_CURSOR;
+import static io.vertx.mysqlclient.impl.protocol.Packets.EnumCursorType.*;
 
 class ExtendedBatchQueryCommandCodec<R> extends ExtendedQueryCommandBaseCodec<R, ExtendedQueryCommand<R>> {
 
   private final List<TupleInternal> params;
-  private int batchIdx = 0;
+  private final boolean[] bindingFailures;
```

Comment:
you can use java.util.BitSet instead, it provides a more compact representation of the same values